### ISSUE TITLE
ParseObject.get has side effects for pending operations on nested fields

### ISSUE THREAD

**Issue by mstniy** [parse-community/Parse-SDK-JS#1365]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description
Pending operations on nested fields of depth at least two get re-done for every call to .get()

### Steps to reproduce

1. Create an object with a double-nested integer field:
```
const obj = new TestObject();
obj.set('objectField', { foo: { bar: 5 } });
```
2. Save the object to create it:
```
await obj.save();
```
3. Increment the nested field:
```
obj.increment('objectField.foo.bar', 15);
```
4. Read its value twice:
```
assert.equal(obj.get('objectField').foo.bar, 20);
assert.equal(obj.get('objectField').foo.bar, 20);
```

### Actual Outcome
The first assert succeeds, the second fails with 35 != 20
Note that without the call to `.save()`, even the first assert fails with 15 != 20, which might be a separate issue.

### Expected Outcome
Both asserts should succeed

###  Environment

Server
- Parse Server version: `4.5.0`
- Operating system: `Ubuntu 20.04`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `v4.4.5`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- Parse JS SDK version: `master` as of 14 May

**Comment by mstniy** [comment]
[These lines](https://github.com/parse-community/Parse-SDK-JS/blob/ffc523f9cd0fa2dd3fcc9754d5f15c3c1c78314f/src/ObjectStateMutations.js#L119-L121) should probably be doing a deep-copy.
Feels like we should also remove line 138 and the spread at line 139.

**Comment by mstniy** [comment]
Fixed after #1451
